adapters/tappx/tappx.go

```diff
@@ -0,0 +1,207 @@
+		errs = append(errs, err)
+		return nil, errs
+	}
+
+	var tappxExt openrtb_ext.ExtImpTappx
+	if err := json.Unmarshal(bidderExt.Bidder, &tappxExt); err != nil { // 2
+		errs = append(errs, err)
+		return nil, errs
+	}
+
+	var test int
+	test = int(request.Test)
+
+	url, err := a.buildEndpointURL(&tappxExt, test)
+	if url == "" {
+		return nil, []error{err}
+	}
+
+	//fmt.Println(url)
+
+	reqJSON, err := json.Marshal(request) // 1
+	if err != nil {
+		errs = append(errs, err)
+		return nil, errs
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	return []*adapters.RequestData{{
+		Method:  "POST",
+		Uri:     url,
+		Body:    reqJSON,
+		Headers: headers,
+	}}, errs
+}
+
+// Builds enpoint url based on adapter-specific pub settings from imp.ext
+func (a *TappxAdapter) buildEndpointURL(params *openrtb_ext.ExtImpTappx, test int) (string, error) {
+	reqHost, reqKey, reqEndpoint := "", "", ""
+	if params.Host != "" {
+		reqHost = params.Host
+	}
+	if params.Endpoint != "" {
+		reqEndpoint = params.Endpoint
+	}
+	if params.TappxKey != "" {
+		reqKey = params.TappxKey
+	}
+
+	if reqHost == "" {
+		return "", &errortypes.BadInput{
+			Message: "Tappx host undefined",
+		}
+	}
+
+	endpointParams := macros.EndpointTemplateParams{Host: reqHost}
+	host, err := macros.ResolveMacros(a.endpointTemplate, endpointParams)
+
+	if err != nil {
+		return "", &errortypes.BadInput{
+			Message: "Unable to parse endpoint url template",
+		}
+	}
+
+	if reqKey == "" {
+		return "", &errortypes.BadInput{
+			Message: "Tappx key undefined",
+		}
+	}
+
+	if reqEndpoint == "" {
+		return "", &errortypes.BadInput{
+			Message: "Tappx endpoint undefined",
+		}
+	}
+
+	thisURI := host + params.Endpoint + "?tappxkey=" + params.TappxKey
+
+	if test == 0 {
+		t := time.Now().UnixNano() / (int64(time.Millisecond) / int64(time.Nanosecond))
+		thisURI = thisURI + "&ts=" + strconv.Itoa(int(t))
+	}
+
+	thisURI = thisURI + "&v=" + TAPPX_BIDDER_VERSION
+
+	//fmt.Println(thisURI)
+
+	return thisURI, nil
+}
+
+func (a *TappxAdapter) MakeBids(internalRequest *openrtb.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if response.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+
+	if response.StatusCode == http.StatusBadRequest {
+		return nil, []error{&errortypes.BadInput{
+			Message: fmt.Sprintf("Unexpected status code: %d. Run with request.debug = 1 for more info", response.StatusCode),
+		}}
+	}
+
+	if response.StatusCode != http.StatusOK {
+		return nil, []error{fmt.Errorf("Unexpected status code: %d. Run with request.debug = 1 for more info", response.StatusCode)}
+	}
+
+	var bidResp openrtb.BidResponse
+	if err := json.Unmarshal(response.Body, &bidResp); err != nil {
+		return nil, []error{err}
+	}
+
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(5)
+
+	var errs []error
```

Comment:
This local variable is not really necessary since the array is not used in any way in the subsequent nested for loop. 
```
182     var errs []error 
```